src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/GlobalSearchResource.kt

```diff
@@ -35,6 +40,15 @@ class GlobalSearchResource(
     @ParameterObject @PageableDefault pageable: Pageable
   ) = globalSearchService.findByGlobalSearchCriteria(globalSearchCriteria, pageable)
 
+  @GetMapping("/prisoner/{id}")
+  @PreAuthorize("hasAnyRole('ROLE_VIEW_PRISONER_DATA', 'ROLE_PRISONER_SEARCH')")
+  @Operation(
+    summary = "Get prisoner by Id",
+    description = "Requires ROLE_PRISONER_SEARCH or ROLE_VIEW_PRISONER_DATA role",
+    security = [SecurityRequirement(name = "ROLE_VIEW_PRISONER_DATA"), SecurityRequirement(name = "ROLE_PRISONER_SEARCH")],
+  )
+  fun findByPrison(@PathVariable id: String) = prisonerIndexService.get(id).takeIf { it != null } ?: throw NotFoundException("$id not found")
+
```

Comment:
Very minor but pretty certain `takeIf { it != null }` is redundant

Reply:
yes!